### PR TITLE
Small stuff

### PR DESCRIPTION
**Pull request by h-2**

### CODE REVIEW

File: src/view_pos_transform.hpp [hunk 1]
```diff
@@ -34,13 +34,13 @@ class view_pos_transform : public ranges::view_base
...
-    std::shared_ptr<pos_transform_t>    pos_transform = nullptr;
-    std::shared_ptr<size_transform_t>   size_transform = nullptr;
+    urng_t                                        urange;
+    ::ranges::semiregular_box_t<pos_transform_t>  pos_transform;
+    ::ranges::semiregular_box_t<size_transform_t> size_transform;
 
     // Associated types iterator.
-    using reference         = decltype((*pos_transform)(urange, 0));
-    using const_reference   = decltype((*pos_transform)(std::as_const(urange), 0));
+    using reference         = decltype(pos_transform(urange, 0));
```
Comment (sarahet) [inline src/view_pos_transform.hpp:42]: I thought it needs to stay :wink:

File: src/mkindex.cpp [hunk 2]
```diff
@@ -231,10 +231,10 @@ void realMain(LambdaIndexerOptions     const & options)
         if constexpr (c_transAlph == AlphabetEnum::AMINO_ACID)
             redSbjSeqs = transSbjSeqs | seqan3::views::deep{seqan3::views::convert<TRedSbjAlph>};
         else if constexpr (c_redAlph == AlphabetEnum::DNA3BS)
-            redSbjSeqs = transSbjSeqs | views::dna_n_to_random<seqan3::dna4>
+            redSbjSeqs = transSbjSeqs | views::dna_n_to_random
```
Comment (sarahet) [inline src/mkindex.cpp:234]: Yes, that is much better

**Review by sarahet — APPROVED** [review]
LGTM

File: src/view_pos_transform.hpp (hunk 1 above)
Comment (h-2) [reply src/view_pos_transform.hpp:42]: Yeah. So. semiregular_box is an addition to C++20. But only *exposion-only* which means that it's behaviour is specified and other components rely on it, but it's not actually there. The way it is specified is that it requires `operator*`, but the version that is in the range-v3 library does not require it 🤷🏻

**Pull request merged**